react/finger-exercises/redux/02 - react-redux/src/app/screens/App/components/ShoppingCart/index.js

```diff
@@ -47,4 +49,19 @@ ShoppingCart.propTypes = {
   removeItem: func.isRequired
 };
 
-export default ShoppingCart;
+const mapStateToProps = state => ({
+  data: state.bookSelected
+});
+
+const mapDispatchToProps = dispatch => {
+  const { addItem, removeItem } = actions;
+  return {
+    addItem: itemId => dispatch(addItem(itemId)),
+    removeItem: item => dispatch(removeItem(item))
+  };
+};
+
+export default connect(
```

Comment:
this component is already receiving the actions and the data props from `App/index.js ` when you do
```
  <ShoppingCart data={bookSelected} addItem={addItem} removeItem={removeItem} />
``` 
so you can skip connecting this component to redux 😛